src/java/picard/sam/SortSam.java

```diff
@@ -51,16 +51,27 @@
 )
 public class SortSam extends CommandLineProgram {
     static final String USAGE_SUMMARY = "Sorts a SAM or BAM file.  ";
-    static final String USAGE_DETAILS = "This tool sorts the input SAM or BAM file by coordinate, queryname  or some other property " +
-            "of the SAMRecord. Input and output formats (SAM or BAM) are determined by the file extension." +
-            "<br />" +
+    static final String USAGE_DETAILS = "This tool sorts the input SAM or BAM file by coordinate, queryname (QNAME), or some other property " +
+            "of the SAM record. The SortOrder of a SAM/BAM file is found in the SAM file header tag (@HD) in the field labeled (SO).  " +
+            "" +
+            "<p>For a coordinate sorted SAM/BAM file, read alignments are sorted first by the reference sequence name (RNAME) field using the " +
+            "reference sequence dictionary (@SQ tag).  Alignments within these subgroups are secondarily sorted using the left-most mapping " +
+            "position of the read (POS).  Subsequent to this sorting scheme, alignments are listed arbitrarily.</p>" +
+            "" +
+            "For QNAME-sorted alignments, all alignments are grouped using the query name (QNAME) field but the alignments are not necessarily sorted within these groups.  " +
+            "Reads having the same QNAME are derived from the same template.  " +
+            "Please be aware that the constituents of a QNAME can vary among different platforms.  For additional information on " +
+            "QNAMES please visit the following <a href='https://www.broadinstitute.org/gatk/guide/article?id=6472'>GATK dictonary</a> entry. " +
+            " Note that for this entry, 'read names' and QNAMES are the same. " +
+
             "<h4>Usage example:</h4>" +
             "<pre>" +
             "java -jar picard.jar SortSam \\<br />" +
             "      I=input.bam \\<br />" +
             "      O=sorted.bam \\<br />" +
             "      SORT_ORDER=coordinate" +
             "</pre>" +
+            "For additional information on the SAM specification, please see this <a href='https://samtools.github.io/hts-specs/SAMv1.pdf'>document</a>.</p>" +
```

Comment:
SAM spec is referred to at a higher level, so I'd rather not list it here, if only because if/when the spec gets updated to a new major version the URL will no longer be correct and no one will remember to update it here.